packages/cli-kit/src/public/node/themes/api.ts

```diff
@@ -77,32 +81,73 @@ export async function fetchChecksums(id: number, session: AdminSession): Promise
   return []
 }
 
-interface UpgradeThemeOptions {
-  fromTheme: number
-  toTheme: number
-  script?: string
-  session: AdminSession
-}
-
-export async function upgradeTheme(upgradeOptions: UpgradeThemeOptions): Promise<Theme | undefined> {
-  const {fromTheme, toTheme, session, script} = upgradeOptions
-  const params = {from_theme: fromTheme, to_theme: toTheme, ...(script && {script})}
-  const response = await request('POST', `/themes`, session, params)
-  return buildTheme(response.json.theme)
-}
+export async function themeUpdate(id: number, params: ThemeParams, session: AdminSession): Promise<Theme | undefined> {
+  const name = params.name
+  if (name === undefined) {
+    throw new AbortError('Theme name is required')
+  }
 
-export async function updateTheme(id: number, params: ThemeParams, session: AdminSession): Promise<Theme | undefined> {
-  const response = await request('PUT', `/themes/${id}`, session, {theme: {id, ...params}})
-  return buildTheme(response.json.theme)
+  const {themeUpdate} = await adminRequestDoc(ThemeUpdate, session, {id: themeGid(id), input: {name}})
+  if (themeUpdate) {
+    const theme = themeUpdate.theme
+    if (theme) {
+      return buildTheme({
+        id: themeGidToId(theme.id),
+        name: theme.name,
+        role: theme.role.toLowerCase(),
+      })
+    } else if (themeUpdate.userErrors.length) {
+      const userErrors = themeUpdate.userErrors.map((error) => error.message).join(', ')
+      throw new AbortError(userErrors)
+    } else {
+      // An unexpected error if neither theme nor userErrors are returned
+      unexpectedGraphQLError('Failed to update theme')
+    }
+  } else {
+    // An unexpected error occured during the GraphQL request execution
+    unexpectedGraphQLError('Failed to update theme')
+  }
 }
 
-export async function publishTheme(id: number, session: AdminSession): Promise<Theme | undefined> {
-  return updateTheme(id, {role: 'main'}, session)
+export async function themePublish(id: number, session: AdminSession): Promise<Theme | undefined> {
```

Comment:
🚲 🖌️ (nit) feel free to ignore completely, just sharing how I might have written this now that I see your new approach!

```ts
export async function themePublish(id: number, session: AdminSession): Promise<Theme | undefined> {
  try {
    const {themePublish} = await adminRequestDoc(ThemePublish, session, {id: themeGid(id)})

    if (!themePublish) throw new Error('')

    const {theme, userErrors} = themePublish

    if (userErrors.length) {
      throw new Error(userErrors.map((error) => error.message).join(', '))
    }

    if (!theme) throw new Error('')

    return buildTheme({
      id: themeGidToId(theme.id),
      name: theme.name,
      role: theme.role.toLowerCase(),
    })
  } catch (error: unknown) {
    if (error instanceof Error) {
      throw new AbortError(error.message)
    }
    throw new AbortError('An unknown error occurred')
  }
}
```

I like the try/catch here because it means we can just `throw` when we don't like the state and we don't end up in nested if/else statements.

Reply:
I used a mix of yours and guilherme's comments. I don't love throwing `Error('')` and then populating it later , but agreed with less if/else statements.